.github/workflows/run_coverage.yml

```diff
@@ -0,0 +1,57 @@
+name: Run Coverage
+on: [pull_request]
+jobs:
+  test:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install poetry
+        run: |
+          pip install poetry
+      - name: Syncify Client
+        run: |
+          python scripts/syncronizer.py
+      - name: Start Database
+        run: |
+          cd api/local
+          docker compose up --wait
+          cd ../..
+      - name: Start Server
+        run: |
+          cd api
+          poetry install --no-root
+          poetry run uvicorn src.main:app &
+          sleep 5
+          cd ..
+        env:
+          DATABASE_TYPE: postgres
+          CONNECTION_URI: postgresql://testuser:testpwd@localhost:5432/honcho
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+      - name: Run Tests
+        run: |
+          cd sdk
+          poetry install
+          poetry run coverage run -m pytest
+          poetry run coverage xml -o coverage.xml
```

Comment:
Don't necessarily need to resolve this if it's not possible but would like a more detailed breakdown of the report if possible. The coverage docs give this as an example: https://nedbatchelder.com/files/sample_coverage_html/index.html